### ISSUE TITLE
Switch to PHP Coding Standards Fixer

### ISSUE THREAD

**Issue by Glavin001** [Glavin001/atom-beautify#175]
See http://cs.sensiolabs.org/

As per @ffsantos92  recommendation ( https://github.com/Glavin001/atom-beautify/issues/125#issuecomment-68526452 ).

**Comment by drawcard** [comment]
This would be awesome, I can't figure out how to get PHP_Beautifier up and running on Mac OSX so this looks like it will be a much easier solution.

**Comment by m1ga** [comment]
any news on this?

**Comment by Glavin001** [comment]
I'm in the middle of exams so it's unlikely I'll get to this soon. I am always open for pull requests. Thanks!

**Comment by m1ga** [comment]
no problem :smile: 

I've tried to add it but I wasn't able to get it running. Basically there isn't much to change cause it should work by calling "php-cs-fixer fix FILE". But I've ended up with

~~~
 /bin/sh -c [ -f ~/.bash_profile ] && source ~/.bash_profile;[ -f ~/.bashrc ] && source ~/.bashrc;php-cs-fixer fix "/tmp/input115111-4110-1hds5wi"
~~~

if I remove the /bin/sh -c its working, but I don't know where its coming from.

Also if I understand your code correctly it will save it to a temp file, apply to beautify and then opens the output-file or stdinput at the end. Since php-cs-fixer will beautify the input file I change outputPath = inputPath. Is that correct?

**Comment by Glavin001** [comment]
> if I remove the /bin/sh -c its working, but I don't know where its coming from.

I don't know about `/bin/sh -c` specifically, I'd suppose that's just how [`exec`](https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L57) for Node.js does it, however the rest of that comes from https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L45-L50

> Also if I understand your code correctly it will save it to a temp file, apply to beautify and then opens the output-file or stdinput at the end. 

You are correct :smiley:.

> Since php-cs-fixer will beautify the input file I change outputPath = inputPath. Is that correct?

`php-cs-fixer` should beautify the input file to another output file, not have side-effects. However, if this is for some reason not yet possible (sounds silly to me) then we will have to rethink how this is performed, because the `inputPath` is [written to with the code to beautify](https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L11-L16) and then [either stdout or the `outputPath` is read as the beautification result](https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee#L62-L76). This is all handled internally with [`cli-beautify` class](https://github.com/Glavin001/atom-beautify/blob/master/lib/langs/cli-beautify.coffee) so there is not really  an option to have `outputPath` = `inputPath`.

I have in mind a lot of changes for how the beautifiers work. More like classes/extended classes that are registered as Beautifiers, and have more control, options, more modular and self contained (no more of having a massive list of settings, etc). Unfortunately, this changes will have to wait until there's time.
Hope we can figure out how to get by `php-cs-fixer` using the `inputPath` instead of outputting to `outputPath`.

**Comment by m1ga** [comment]
Thanks for the detailed explanation

I don't see an option for a different output file or stdout in the config settings of the cs page. Thats why I was confused.

https://www.npmjs.com/package/exec states that it uses -c but it gives me this error "-f: Zeile 0: [: Fehlende `]' " (Row 0: [: missing ']")
If I run the command by hand its working. atom-beautify removes the input file after the error and then php-cs-fixer can't find the tmp file.

If I remove  "/bin/sh -c" and just leave the rest its working fine (from command line)

I'll have a look at it again at the weekend. Perhaps I can get it running.

**Comment by m1ga** [comment]
stdout is not supported
https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/784
https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/454
and I don't see a switch to output it to a different file

Also it only works for the php code. If you have some html it won't indent it:
http://stackoverflow.com/questions/27961446/psr-2-compatible-inline-php-tags-in-html-templates

About the error message:
the php-cs-fixer is working! When I switch off the delete function I can see, that the inputFile (is used this for testing) is correct. But I'm not sure why its going into the "unless err" (cli-beautify.coffee) function, because err is empty.

**Comment by drawcard** [comment]
Damn. For what it's worth I have been looking into beautifying files using Grunt / Gulp task managers, as a stop gap until I can do it inside the IDE.

**Comment by Glavin001** [comment]
Closed thanks to #242.

Published to v0.23.0
